src/extension.ts

```diff
@@ -29,43 +41,94 @@ const extract = async (options: {
     )
 
     const text = document.getText()
-    const { unbound, styledImported } = collectUnbound(text)
+    const unbound = collectUnbound(text)
     if (!unbound.length) {
       vscode.window.showWarningMessage(
-        '[SCE] Nothing to copy: There is no unbound components',
+        '[SCE] Nothing to extract: There are no unbound components',
       )
       return
     }
+
+    const exportIdentifier =
+      type == 'extractExportedToClipboard' || type == 'extractToSeparateFile'
+
     const declarations = generateDeclarations({
       unbound,
-      styledImported,
-      exportIdentifier: options.exportIdentifier,
-      importStyled: config.get('addImportStatement', true),
+      exportIdentifier,
     })
 
-    await vscode.env.clipboard.writeText(declarations)
-    vscode.window.showInformationMessage(
-      `[SCE] Copied to clipboard! (Found: ${unbound.length}) `,
-    )
+    if (type == 'extractToClipboard' || type == 'extractExportedToClipboard') {
+      let clipboardText = declarations
+      if (config.get('addImportStatement', true)) {
+        const styledImportInsertion = getStyledImportInsertion(
+          editor.document.getText(),
+        )
+        if (styledImportInsertion) {
+          clipboardText = styledImportInsertion.insertionText + declarations
+        }
+      }
+
+      await vscode.env.clipboard.writeText(clipboardText)
+
+      vscode.window.showInformationMessage(
+        `[SCE] Copied to clipboard! (Found: ${unbound.length}) `,
+      )
+    } else if (type == 'extractToSeparateFile') {
+      const styleFile = getCorrespondingStyleFile(
+        editor.document.uri.path,
+        config.get('separateFile.matchPattern', '*'),
+        config.get('separateFile.outputFile', 'styles.ts'),
+      )
+      if (!styleFile) {
+        vscode.window.showWarningMessage(
+          '[SCE] This file does not match the pattern in your configuration.',
+        )
+        return
+      }
+
+      await modifyImports(editor, styleFile, unbound)
+
+      const styleFileEditor = await openFileInEditor(styleFile)
+      await insertStyles(styleFileEditor, declarations)
+      await insertStyledImport(styleFileEditor)
+
+      await editor.document.save()
+      await styleFileEditor.document.save()
```

Comment:
1. Get the corresponding style file
2. Edit the imports of the current file to point to the corresponding style file
3. Open the style file
4. Insert the generated styles
5. Insert the `styled` import if not already imported
6. Save both files.